latex-syntax/src/main/java/org/xwiki/contrib/latex/internal/FigureTool.java

```diff
@@ -42,4 +49,60 @@
      * @return true if the figure caption macro is the last content inside the figure macro content or false otherwise
      */
     boolean isFigureCaptionLast(Block figureCaptionBlock);
+
+    /**
+     * Returns the environment to use for the current caption. By default, "table" or "figure" are returned but
+     * extensions can allow for different environments. Note that different environments mights require different
+     * parameters (see {@link #getFigureParameter(FigureBlock)}), or might not support captions in their content (see
+     * {@link #displayFigureCaption(FigureCaptionBlock)}).
+     *
+     * @param figureBlock the figure block for which to get the environment name
+     * @return the name of the environment (e.g., "figure", "block", or another value in case of customization by an
+     *     extension)
+     * @since 1.21
+     */
+    @Unstable
+    default String getFigureEnvironment(Block figureBlock)
+    {
+        return isTable(figureBlock) ? "table" : "figure";
+    }
+
+    /**
+     * Return the figure parameter for a figure block. The figure parameter must be synchronized with the returned
+     * figure environment (see {@link #getFigureEnvironment(Block)}. For instance, for the figure environments "figure"
+     * or "table", the default parameters is {@code "h"} (setting the figure location to "here"), resulting in the
+     * following latex.
+     * <pre>
+     * \begin{figure}[h]
+     * % Figure content.
+     * \end{figure}
+     * </pre>
```

Comment:
what will be the parameter used for theorems?

Reply:
Either no parameter when the theorem for no have a caption
```latex
\begin{theorem}
% ...
\end{theorem}
```

Or the caption text if the figure contains a captions.
```latex
\begin{theorem}[Caption text]
% ...
\end{theorem}
```